airbyte-integrations/connectors/source-snowflake/source_snowflake/authenticator.py

```diff
@@ -0,0 +1,78 @@
+import base64
+import hashlib
+from datetime import timedelta, datetime, timezone
+from typing import Text, Mapping, Any
+
+from airbyte_cdk.sources.declarative.auth import JwtAuthenticator
+from airbyte_cdk.sources.declarative.auth.jwt import JwtAlgorithm
+from cryptography.hazmat.primitives._serialization import Encoding, PublicFormat
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
+from cryptography.hazmat.backends import default_backend
+
+
+class SnowflakeJwtAuthenticator(JwtAuthenticator):
+    TOKEN_DURATION = 55
+    SNOWFLAKE_URL_SUFFIX = ".snowflakecomputing.com"
+    HTTP_PREFIX = "https://"
+
+    def get_auth_header(self) -> Mapping[str, Any]:
+        """
+        :return: A dictionary containing all the necessary headers to authenticate.
+        """
+        return {
+            'Authorization': f"Bearer {self.token}"
+        }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @classmethod
+    def from_config(cls, config):
+        sub = cls.get_formatted_account(config['host'], config["credentials"]['user_name'])
+        kwargs = {"config": {},
+                  "algorithm": JwtAlgorithm.RS256,
+                  "iss": cls.calculate_public_key_fingerprint(config["credentials"]['private_key'], sub),
```

Comment:
It seems odd to me that we need the private key here. My understanding of JWT is that the private key is used to decrypt the data. As part of "logging in", only the public key should be needed. Can you explain why we can't use the public key here?